packages/manager/modules/hub/src/components/billing-summary/controller.js

```diff
@@ -8,46 +8,57 @@ export default class ManagerHubBillingSummaryCtrl {
     this.$translate = $translate;
     this.atInternet = atInternet;
     this.coreURLBuilder = coreURLBuilder;
+    this.loading = true;
+    this.billsError = false;
   }
 
   $onInit() {
     this.DEBT_PAY_URL = this.coreURLBuilder.buildURL(
       'dedicated',
       '#/billing/history/debt/all/pay',
     );
-    this.loading = true;
-    const loadBills = this.$q
-      .when(this.bills ? this.bills : this.fetchBills())
-      .then(({ data }) => {
-        this.bills = data;
-        this.formattedBillingPrice = this.getFormattedPrice(
-          data.total,
-          get(data, 'currency.code'),
-        );
-        this.buildPeriodFilter(data.period);
-        return this.bills;
-      });
-    const loadDebt = this.$q
-      .when(this.debt ? this.debt : this.fetchDebt())
-      .then(({ data }) => {
-        this.debt = data;
-        this.formattedDebtPrice = this.getFormattedPrice(
-          get(data, 'dueAmount.value'),
-          get(data, 'dueAmount.currencyCode'),
-        );
-        return this.debt;
-      });
-
     this.$translate.refresh().then(() => {
       this.periods = [1, 3, 6].map((months) => ({
         value: months,
         label: this.$translate.instant(`hub_billing_summary_period_${months}`),
       }));
       [this.billingPeriod] = this.periods;
     });
+    this.loadAll();
+  }
 
-    return this.$q.all([loadBills, loadDebt]).finally(() => {
-      this.loading = false;
+  loadAll() {
+    this.loading = true;
+    this.$q
+      .all([this.loadBills(), this.loadDebt()])
+      .catch(() => {
+        this.billsError = true;
+      })
+      .finally(() => {
+        this.loading = false;
+      });
+  }
+
+  loadBills() {
+    return this.fetchBills().then(({ data }) => {
+      this.bills = data;
+      this.formattedBillingPrice = this.getFormattedPrice(
+        data.total,
+        get(data, 'currency.code'),
```

Comment:
nitpick: we could
```suggestion
        data?.currency?.code,
```

Reply:
We could, i'll check if rollup is configured to accept this, if it is, i'll change it everywhere. :)

Reply:
@zakariasahmane did you figure out if rollup handles it ?

Reply:
Hi @marie-j !

Sorry i didn't give feedback ! Actually it looks like it's handled ( i tested it for this module and everything works just fine ), but instead of removing it just here, it will be the subject of another PR to remove lodash-es when not necessary, there are a lot of packages that could benefit from removing it.